orchestrator/crates/orchestrator/src/core/config.rs

```diff
@@ -0,0 +1,333 @@
+    pub async fn from_run_cmd(run_cmd: &RunCmd) -> OrchestratorResult<Self> {
+        let cloud_provider = CloudProvider::try_from(run_cmd.clone())?;
+        let provider_config = Arc::new(cloud_provider);
+
+        let db: DatabaseArgs = DatabaseArgs::try_from(run_cmd.clone())?;
+        let storage_args: StorageArgs = StorageArgs::try_from(run_cmd.clone())?;
+        let alert_args: AlertArgs = AlertArgs::try_from(run_cmd.clone())?;
+        let queue_args: QueueArgs = QueueArgs::try_from(run_cmd.clone())?;
+
+        let prover_config = ProverConfig::try_from(run_cmd.clone())?;
+        let da_config = DAConfig::try_from(run_cmd.clone())?;
+        let settlement_config = SettlementConfig::try_from(run_cmd.clone())?;
+
+        let params = ConfigParam {
+            madara_rpc_url: run_cmd.madara_rpc_url.clone(),
+            snos_config: SNOSParams::from(run_cmd.snos_args.clone()),
+            service_config: ServiceParams::from(run_cmd.service_args.clone()),
+            server_config: ServerParams::from(run_cmd.server_args.clone()),
+            snos_layout_name: Self::get_layout_name(run_cmd.proving_layout_args.prover_layout_name.clone().as_str())?,
+            prover_layout_name: Self::get_layout_name(run_cmd.proving_layout_args.snos_layout_name.clone().as_str())?,
+        };
+        let rpc_client = JsonRpcClient::new(HttpTransport::new(params.madara_rpc_url.clone()));
+
+        let mut processing_locks = ProcessingLocks::default();
+
+        if let Some(max_concurrent_snos_jobs) = params.service_config.max_concurrent_snos_jobs {
+            processing_locks.snos_job_processing_lock =
+                Some(Arc::new(JobProcessingState::new(max_concurrent_snos_jobs)));
+        }
+
+        if let Some(max_concurrent_proving_jobs) = params.service_config.max_concurrent_proving_jobs {
+            processing_locks.proving_job_processing_lock =
+                Some(Arc::new(JobProcessingState::new(max_concurrent_proving_jobs)));
+        }
+
+        let database = Self::build_database_client(&db).await?;
+        let storage = Self::build_storage_client(&storage_args, provider_config.clone()).await?;
+        let alerts = Self::build_alert_client(&alert_args, provider_config.clone()).await?;
+        let queue = Self::build_queue_client(&queue_args, provider_config.clone()).await?;
+
+        // External Clients Initialization
+        let prover_client = Self::build_prover_service(&prover_config, &params);
+        let da_client = Self::build_da_client(&da_config).await;
+        let settlement_client = Self::build_settlement_client(&settlement_config).await?;
+
+        Ok(Self {
+            params,
+            madara_client: Arc::new(rpc_client),
+            database,
+            storage,
+            alerts,
+            queue,
+            prover_client,
+            da_client,
+            processing_locks,
+            settlement_client,
+        })
+    }
+
+    pub(crate) async fn build_database_client(
```

Comment:
Maybe we should rename this to be more clear about the fact that this initializes mongodb under the hood for the time being.
```suggestion
    pub(crate) async fn build_database_client_mongo(
```

Reply:
@Trantorian1 don't able to understand this

Reply:
Nah no worries this was just a nitpick anyway X)